packages/firestore/src/api/database.ts

```diff
@@ -1478,10 +1346,8 @@ export class DocumentSnapshot<T = DocumentData>
 
   get(
     fieldPath: string | ExternalFieldPath,
-    options?: PublicSnapshotOptions
+    options: PublicSnapshotOptions = {}
```

Comment:
Does this require an API change to the d.ts file, or is this rendered by tsc as a local assignment if not defined? Stated alternatively: does the caller supply the default argument or does the callee?

Reply:
The default argument is provided by the callee. An example in ES3:

```
function foo(bar = {}) {
}
```

transpiles to:

```
"use strict";
function foo(bar) {
    if (bar === void 0) { bar = {}; }
}
 ```

ES5+JavaScript supports default arguments, but the semantics match the code snippet above.